### PR TITLE
Qs 72 - add event rule to trigger deletion of old job_status rows

### PR DESCRIPTION
**Pull request by MDunitz**
# Need
- a way to remove old jobs (+90 days) from the db

# approach
- lambda triggered by an AWS event calls `job_status.delete_old_rows()`

# Tests
- [x] tested in predev env, successfully deleted only old jobs

# Questions
- generalize `delete_all_rows` to take a number of days as an argument to make it more flexible/reusable?

### CODE REVIEW

**Comment by kislyuk** [comment]
@MDunitz could you please give this PR a more descriptive title?

File: .gitignore [hunk 1]
```diff
@@ -37,4 +37,6 @@ load_data_lambda/vendor/*
 target/
 load_data.zip
 create_async_query.zip
+db_cleanup.zip
```
Comment (mweiden) [inline .gitignore]: how about just `*.zip`?

File: db_cleanup_lambda/Makefile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+LAMBDA_NAME=db_cleanup
+ZIP_FILE=$(LAMBDA_NAME).zip
+BUCKET=query-service-$(AWS_ACCOUNT_ID)
+STAGED_FILE_KEY=$(DEPLOYMENT_STAGE)/lambda_deployments/$(LAMBDA_NAME)/$(ZIP_FILE)
+
+default: build
+
+
+
```
Comment (mweiden) [inline db_cleanup_lambda/Makefile]: rm whitespace

File: .gitignore (hunk 1 above)
Comment (mweiden) [reply .gitignore]: also: you did check in `db_cleanup_lambda/db_cleanup.zip`. rm this

File: db_cleanup_lambda/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+dcplib==1.4.0
+inflection==0.3.1
```
Comment (mweiden) [inline db_cleanup_lambda/requirements.txt]: why do you need `inflection` for the db_cleanup script?

File: query/test/unit/db_cleanup_lambda/test_db_cleanup.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        assert mock_job_status.delete_old_rows.called_once()
+
+    def test_db_contains_no_job_status_rows_older_than_90_days(self):
+        db = PostgresDatabase(Config.serve_database_uri)
+        with db.transaction() as (_, tables):
+            tables.job_status._cursor.execute(
+                "SELECT count(*) FROM job_status WHERE created_at < NOW() - INTERVAL '91 days'"
+            )
+            count = tables.job_status._cursor.fetchall()
+            assert count[0][0] == 0
```
Comment (mweiden) [inline query/test/unit/db_cleanup_lambda/test_db_cleanup.py]: Was anything ever inserted in the table to check that this does, in fact, delete something?

File: terraform/modules/query-service-infra/db_cleanup_lambda.tf [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    {
+      "Sid": "LambdaLogging",
+      "Action": [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents",
+        "logs:DescribeLogStreams"
+      ],
+      "Resource": [
+        "arn:aws:logs:*:*:*"
```
Comment (mweiden) [inline terraform/modules/query-service-infra/db_cleanup_lambda.tf]: Please limit this just to the required log group

**Comment by mweiden** [comment]
@MDunitz looking like a clean PR. Just a couple neat and tidy things here.

File: query/test/unit/db_cleanup_lambda/test_db_cleanup.py (hunk 4 above)
Comment (MDunitz) [reply query/test/unit/db_cleanup_lambda/test_db_cleanup.py]: yes. I set the created at on a few rows to be now() - 95 days and they were deleted

File: .gitignore (hunk 1 above)
Comment (MDunitz) [reply .gitignore]: we need to keep the psycopg2.zip file

File: db_cleanup_lambda/requirements.txt (hunk 3 above)
Comment (MDunitz) [reply db_cleanup_lambda/requirements.txt]: removed

File: terraform/modules/query-service-infra/db_cleanup_lambda.tf (hunk 5 above)
Comment (MDunitz) [reply terraform/modules/query-service-infra/db_cleanup_lambda.tf]: 👍

File: query/test/unit/db_cleanup_lambda/test_db_cleanup.py (hunk 4 above)
Comment (MDunitz) [reply query/test/unit/db_cleanup_lambda/test_db_cleanup.py]: also I added a test specifically for that

File: .gitignore (hunk 1 above)
Comment (mweiden) [reply .gitignore]: ah ok. You can also `git add -f` that btw.

**Comment by mweiden** [comment]
@MDunitz looks good, seems that you have a linter error, though. I'm good with merging these changes once you get this green.

File: .gitignore (hunk 1 above)
Comment (MDunitz) [reply .gitignore]: 👍

**Pull request merged**